Add minimal details required to successfully Export to Android (#3066)
godotengine/godot-docs: getting_started/workflow/export/exporting_for_android.rst

.. _doc_exporting_for_android:

> 

Reviewer: This extraneous space is not necessary.

.. _doc_exporting_for_android:

> **This page covers the basics** for Android export.  For a more comprehensive walkthrough (*using Godot 3.1*), please [see this excelent tutorial](https://godotlearn.com/godot-3-1-how-to-export-for-android/).

Reviewer 2: Is this part really necessary? I'd rather improve the docs we currently have rather than linking to many external sites.

Author: it's very well written and includes additional info.  but ok not to link  (it's details are beyond the minimum)

Download and install the Android SDK from
https://developer.android.com/studio/

> - You need to run Android Studio once to complete the setup.

Reviewer: There's no need to make everything lists, this should not start with `- `.

That's also not really accurate as you don't need Android Studio to export for Android, only the SDK.

Author: it defaulted me to install Android Studio, updated docs to clarify

Install OpenJDK or Oracle JDK
-----------------------------

Download and install  `OpenJDK <https://github.com/ojdkbuild/ojdkbuild>`__ or `Oracle JDK <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`__. Versions below JDK 8 may not work; some users have reported issues with the jarsigner (used to sign the APKs) in JDK 7.

> - If you install OpenJDK, choose ``1.8``. Don't choose the ``openjdk-jre`` files as that only contains the JRE, not the JDK which is required here.

Reviewer: Doesn't have to be a list.

Download and install the Android SDK from
https://developer.android.com/studio/

- You need to run Android Studio once to complete the setup.

Install OpenJDK or Oracle JDK
-----------------------------

Download and install  `OpenJDK <https://github.com/ojdkbuild/ojdkbuild>`__ or `Oracle JDK <http://www.oracle.com/technetwork/java/javase/downloads/index.html>`__. Versions below JDK 8 may not work; some users have reported issues with the jarsigner (used to sign the APKs) in JDK 7.

> - If you install OpenJDK, choose ```1.8```. and choose do not choose the ```openjdk-jre``` files as that is only the JRE, not the JDK.

Reviewer 2: ```suggestion
- If you install OpenJDK, choose ``1.8``. Don't choose the ``openjdk-jre`` files as that only contains the JRE, not the JDK which is required here.
```

Author: looks good, I don't know how to apply your suggested change to my pull request.  maybe I cannot (only members of godot-docs can?)

    keytool -keyalg RSA -genkeypair -alias androiddebugkey -keypass android -keystore debug.keystore -storepass android -dname "CN=Android Debug,O=Android,C=US" -validity 9999

> - This will create a ``debug.keystore`` file in your current directory. You should move it to a memorable location such as ``%USERPROFILE%\.android\``, because you will need it's location in a later step.

Reviewer: Doesn't have to be a list.

Reviewer: Typo: "it's" -> "its".

>   - see this Q&A for more info on the KeyTool process: https://godotengine.org/qa/21349/jdk-android-file-missing?show=21399#a21399

Reviewer: I think it would be better to integrate the relevant points from this QA answer directly in the docs. It mostly duplicates the existing content but makes it more Windows-friendly, so this should be changed in the main doc to make it friendly to all platforms.

Author: the Q&A has more detail than minimally required, so updating to say that.

    keytool -keyalg RSA -genkeypair -alias androiddebugkey -keypass android -keystore debug.keystore -storepass android -dname "CN=Android Debug,O=Android,C=US" -validity 9999

> - this will create a ```debug.keystore-``` file in your current directory.  you should move it to ```%userprofile%/.android/```

Reviewer 2: ```suggestion
- This will create a ``debug.keystore`` file in your current directory. You should move it to ``%USERPROFILE%\.android\``.
```

Also, is it actually necessary to move this file? As far as I know, it can be located anywhere.

Maybe Godot could be modified to use a default keystore path, which would remove the need to configure the debug keystore path in the Editor Settings.

Author: you are right, the ``debug.keystore`` can be anywhere, the user just needs to remember the location for a further step.   I don't know how to apply your edit, so maybe you can add that "remember the location" details.

In that screen, the path to 3 files needs to be set:

-  The *adb* executable (adb.exe on Windows)
  - It can usually be found at ``%LOCALAPPDATA%\Android\Sdk\platform-tools\adb.exe``.
-  The *jarsigner* executable (from JDK 6 or 8)

>   - On Windows, OpenJDK installs to ``%PROGRAMFILES%\ojdkbuild\java-1.8.0-openjdk-1.8.0.232-2\bin``. The exact path may vary depending on the OpenJDK update you've installed.

Reviewer: That's too specific. It depends on what distribution you used for OpenJDK, what version you used, etc.
It should simply mention that it should be in the `bin` directory of the JDK installation.

Author: i don't know how to say that tersely yet be obvious to noobs what the folder would look like, so I think this is a good compromise.

In that screen, the path to 3 files needs to be set:

-  The *adb* executable (adb.exe on Windows)

>   - This will usually be found at ```%appdata%/../local/Android/Sdk/platform-tools/adb.exe```

Reviewer 2: ```suggestion
  - It can usually be found at ``%LOCALAPPDATA%\Android\Sdk\platform-tools\adb.exe``.
```

Author: looks good.   sorry for all the grammar / style errors :(

-  The *jarsigner* executable (from JDK 6 or 8)

>   - on windows, OpenJDK installs to ```%programfiles%/ojdkbuild\java-1.8.0-openjdk-1.8.0.232-2\bin```

Reviewer 2: ```suggestion
  - On Windows, OpenJDK installs to ``%PROGRAMFILES%\ojdkbuild\java-1.8.0-openjdk-1.8.0.232-2\bin``. The exact path may vary depending on the OpenJDK update you've installed.
```

-  The debug *keystore*

>   - where you put the ```debug.keystore``` file you created above

Reviewer 2: ```suggestion
  - It can be found in the folder where you put the ``debug.keystore`` file you created above.
```

Author: looks good

You can optimize the size further by compiling an Android export template with
only the features you need. See :ref:`doc_optimizing_for_size` for more
information.

Further reading
---------------

> - Article covering the Android export process [at GodotLearn.com](https://godotlearn.com/godot-3-1-how-to-export-for-android/) (*Uses Godot v3.1*).

Reviewer: The link format is wrong, it should be of this form:
```
`at GodotLearn.com <https://godotlearn.com/godot-3-1-how-to-export-for-android/>`_
```

However, apart from additional screenshots and some more details on specific steps, this external documentation is mostly redundant with this page, and we have no control over it so it might well become outdated as Godot evolves (the export workflow for 3.2 already changed compared to 3.1.2).

So I'd prefer that we improve this page to make it exhaustive instead of referencing thirdparty content.
We do refer to thirdparty content on topics which are not Godot-specific, e.g. matrix calculus, etc., but for Godot's export workflow, there's no reason for the official documentation not to be the reference.

Author: That article has  more details on things beyond the minimum, so I think it would be good, but you and another maintainer would prefer no link so I'll just remove it.

> 

Reviewer: Extra newlines are not necessary.
